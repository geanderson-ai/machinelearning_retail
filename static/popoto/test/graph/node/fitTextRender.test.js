import fitTextRenderer from '../../../src/graph/node/fitTextRenderer'
import provider from '../../../src/provider/provider';
import * as d3 from 'd3';
import {measureTextWidth, getLines} from '../../../src/graph/node/fitTextRenderer'

describe("measureTextWidth", function () {
    test("Should measure undefined text correctly", () => {
        expect(measureTextWidth()).toMatchSnapshot();
    });

    test("Should measure empty text correctly", () => {
        expect(measureTextWidth("")).toMatchSnapshot();
    });

    test("Should measure text correctly", () => {
        expect(measureTextWidth("mockedTextValue")).toMatchSnapshot();
    });
});

describe("getLines", function () {

    test("Should works with undefined as text", () => {
        const lines = getLines();
        expect(lines).toMatchSnapshot();
    });

    test("Should works with empty string as text", () => {
        const lines = getLines("");
        expect(lines).toMatchSnapshot();
    });

    test("Should works with simple word as text", () => {
        const lines = getLines("mockedTextValue");
        expect(lines).toMatchSnapshot();
    });

    test("Should works with multiple words on same line", () => {
        const lines = getLines("mocked text a value");
        expect(lines).toMatchSnapshot();
    });

    test("Should works with multiple words as text", () => {
        const lines = getLines("mocked text value");
        expect(lines).toMatchSnapshot();
    });

    test("Should works with multiple spaces and multiple words as text", () => {
        const lines = getLines("        mocked   text       value           ");
        expect(lines).toMatchSnapshot();
    });

    test("Should works with integer as text", () => {
        const lines = getLines(0);
        expect(lines).toMatchSnapshot();
    });

});

describe("render", function () {

    const cssClassMockValue = "mocked_css_class_value";
    const colorMockValue = "mocked_color_value";

    describe('Should works with undefined text', () => {

        beforeEach(() => {
            document.body.innerHTML =
                '<div>' +
                '<g>' +
                '</g>' +
                '</div>';

            provider.node.getTextValue = jest.fn().mockReturnValueOnce();
            provider.node.getSize = jest.fn().mockReturnValue(50);
            provider.node.getCSSClass = jest.fn().mockReturnValueOnce(cssClassMockValue).mockReturnValueOnce(cssClassMockValue);
            provider.node.getColor = jest.fn().mockReturnValue(colorMockValue);
            fitTextRenderer.getNodeBoundingBox = jest.fn().mockReturnValue({x: -50, y: -50, width: 0, height: 0});
            fitTextRenderer.render(d3.selectAll('g').data([{}]));
        });

        test('Should render correctly', () => {
            expect(document.body.innerHTML).toMatchSnapshot();
        });
    });

    describe('Should works with empty text', () => {

        beforeEach(() => {
            document.body.innerHTML =
                '<div>' +
                '<g>' +
                '</g>' +
                '</div>';

            provider.node.getTextValue = jest.fn().mockReturnValueOnce("");
            provider.node.getSize = jest.fn().mockReturnValue(50);
            provider.node.getCSSClass = jest.fn().mockReturnValueOnce(cssClassMockValue).mockReturnValueOnce(cssClassMockValue);
            provider.node.getColor = jest.fn().mockReturnValue(colorMockValue);
            fitTextRenderer.getNodeBoundingBox = jest.fn().mockReturnValue({x: -50, y: -50, width: 0, height: 0});
            fitTextRenderer.render(d3.selectAll('g').data([{}]));
        });

        test('Should render correctly', () => {
            expect(document.body.innerHTML).toMatchSnapshot();
        });
    });

    describe('Should works with 1 word as text', () => {

        beforeEach(() => {
            document.body.innerHTML =
                '<div>' +
                '<g>' +
                '</g>' +
                '</div>';

            provider.node.getTextValue = jest.fn().mockReturnValueOnce("mockedTextValue");
            provider.node.getSize = jest.fn().mockReturnValue(50);
            provider.node.getCSSClass = jest.fn().mockReturnValueOnce(cssClassMockValue).mockReturnValueOnce(cssClassMockValue);
            provider.node.getColor = jest.fn().mockReturnValue(colorMockValue);
            fitTextRenderer.getNodeBoundingBox = jest.fn().mockReturnValue({x: -50, y: -50, width: 100, height: 100});
            fitTextRenderer.render(d3.selectAll('g').data([{}]));
        });

        test('Should render correctly', () => {
            expect(document.body.innerHTML).toMatchSnapshot();
        });
    });

    describe('Should works with multiple words as text', () => {

        beforeEach(() => {
            document.body.innerHTML =
                '<div>' +
                '<g>' +
                '</g>' +
                '</div>';

            provider.node.getTextValue = jest.fn().mockReturnValueOnce("mocked text value");
            provider.node.getSize = jest.fn().mockReturnValue(50);
            provider.node.getCSSClass = jest.fn().mockReturnValueOnce(cssClassMockValue).mockReturnValueOnce(cssClassMockValue);
            provider.node.getColor = jest.fn().mockReturnValue(colorMockValue);
            fitTextRenderer.getNodeBoundingBox = jest.fn().mockReturnValue({x: -50, y: -50, width: 100, height: 100});
            fitTextRenderer.render(d3.selectAll('g').data([{}]));
        });

        test('Should render correctly', () => {
            expect(document.body.innerHTML).toMatchSnapshot();
        });
    });

    describe('Should works with multiple words on same span', () => {

        beforeEach(() => {
            document.body.innerHTML =
                '<div>' +
                '<g>' +
                '</g>' +
                '</div>';

            provider.node.getTextValue = jest.fn().mockReturnValueOnce("mocked text a value");
            provider.node.getSize = jest.fn().mockReturnValue(50);
            provider.node.getCSSClass = jest.fn().mockReturnValueOnce(cssClassMockValue).mockReturnValueOnce(cssClassMockValue);
            provider.node.getColor = jest.fn().mockReturnValue(colorMockValue);
            fitTextRenderer.getNodeBoundingBox = jest.fn().mockReturnValue({x: -50, y: -50, width: 100, height: 100});
            fitTextRenderer.render(d3.selectAll('g').data([{}]));
        });

        test('Should render correctly', () => {
            expect(document.body.innerHTML).toMatchSnapshot();
        });
    });

    describe('Should works with integer as text', () => {

        beforeEach(() => {
            document.body.innerHTML =
                '<div>' +
                '<g>' +
                '</g>' +
                '</div>';

            provider.node.getTextValue = jest.fn().mockReturnValueOnce(0);
            provider.node.getSize = jest.fn().mockReturnValue(50);
            provider.node.getCSSClass = jest.fn().mockReturnValueOnce(cssClassMockValue).mockReturnValueOnce(cssClassMockValue);
            provider.node.getColor = jest.fn().mockReturnValue(colorMockValue);
            fitTextRenderer.getNodeBoundingBox = jest.fn().mockReturnValue({x: -50, y: -50, width: 100, height: 100});
            fitTextRenderer.render(d3.selectAll('g').data([{}]));
        });

        test('Should render correctly', () => {
            expect(document.body.innerHTML).toMatchSnapshot();
        });
    });

});