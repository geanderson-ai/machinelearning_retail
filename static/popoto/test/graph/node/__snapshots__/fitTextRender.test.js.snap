// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getLines Should works with empty string as text 1`] = `
Object {
  "lines": Array [],
  "textRadius": 0,
}
`;

exports[`getLines Should works with integer as text 1`] = `
Object {
  "lines": Array [
    Object {
      "linesLength": 1,
      "text": "0",
    },
  ],
  "textRadius": 8.48528137423857,
}
`;

exports[`getLines Should works with multiple spaces and multiple words as text 1`] = `
Object {
  "lines": Array [
    Object {
      "linesLength": 3,
      "text": "mocked",
    },
    Object {
      "linesLength": 3,
      "text": "text",
    },
    Object {
      "linesLength": 3,
      "text": "value",
    },
  ],
  "textRadius": 40.24922359499622,
}
`;

exports[`getLines Should works with multiple words as text 1`] = `
Object {
  "lines": Array [
    Object {
      "linesLength": 3,
      "text": "mocked",
    },
    Object {
      "linesLength": 3,
      "text": "text",
    },
    Object {
      "linesLength": 3,
      "text": "value",
    },
  ],
  "textRadius": 40.24922359499622,
}
`;

exports[`getLines Should works with multiple words on same line 1`] = `
Object {
  "lines": Array [
    Object {
      "linesLength": 3,
      "text": "mocked",
    },
    Object {
      "linesLength": 3,
      "text": "text",
    },
    Object {
      "linesLength": 3,
      "text": "a value",
    },
  ],
  "textRadius": 45.69463863518345,
}
`;

exports[`getLines Should works with simple word as text 1`] = `
Object {
  "lines": Array [
    Object {
      "linesLength": 1,
      "text": "mockedTextValue",
    },
  ],
  "textRadius": 90.19977827023745,
}
`;

exports[`getLines Should works with undefined as text 1`] = `
Object {
  "lines": Array [],
}
`;

exports[`measureTextWidth Should measure empty text correctly 1`] = `0`;

exports[`measureTextWidth Should measure text correctly 1`] = `180`;

exports[`measureTextWidth Should measure undefined text correctly 1`] = `0`;

exports[`render Should works with 1 word as text Should render correctly 1`] = `"<div><g><rect fill=\\"mocked_color_value\\" class=\\"mocked_css_class_value\\" x=\\"-53\\" y=\\"-50\\" rx=\\"5\\" ry=\\"5\\" width=\\"106\\" height=\\"100\\" transform=\\"translate(0,0) scale(0.5543250876981161)\\"></rect><text class=\\"mocked_css_class_value\\" style=\\"text-anchor: middle; font: 10px sans-serif\\" transform=\\"translate(0,0) scale(0.5543250876981161)\\"><tspan x=\\"0\\" y=\\"3.6000000000000005\\">mockedTextValue</tspan></text></g></div>"`;

exports[`render Should works with empty text Should render correctly 1`] = `"<div><g><rect fill=\\"mocked_color_value\\" class=\\"mocked_css_class_value\\" x=\\"-53\\" y=\\"-50\\" rx=\\"5\\" ry=\\"5\\" width=\\"6\\" height=\\"0\\" transform=\\"translate(0,0) scale(1)\\"></rect><text class=\\"mocked_css_class_value\\" style=\\"text-anchor: middle; font: 10px sans-serif\\" transform=\\"translate(0,0) scale(1)\\"></text></g></div>"`;

exports[`render Should works with integer as text Should render correctly 1`] = `"<div><g><rect fill=\\"mocked_color_value\\" class=\\"mocked_css_class_value\\" x=\\"-53\\" y=\\"-50\\" rx=\\"5\\" ry=\\"5\\" width=\\"106\\" height=\\"100\\" transform=\\"translate(0,0) scale(5.892556509887896)\\"></rect><text class=\\"mocked_css_class_value\\" style=\\"text-anchor: middle; font: 10px sans-serif\\" transform=\\"translate(0,0) scale(5.892556509887896)\\"><tspan x=\\"0\\" y=\\"3.6000000000000005\\">0</tspan></text></g></div>"`;

exports[`render Should works with multiple words as text Should render correctly 1`] = `"<div><g><rect fill=\\"mocked_color_value\\" class=\\"mocked_css_class_value\\" x=\\"-53\\" y=\\"-50\\" rx=\\"5\\" ry=\\"5\\" width=\\"106\\" height=\\"100\\" transform=\\"translate(0,0) scale(1.2422599874998832)\\"></rect><text class=\\"mocked_css_class_value\\" style=\\"text-anchor: middle; font: 10px sans-serif\\" transform=\\"translate(0,0) scale(1.2422599874998832)\\"><tspan x=\\"0\\" y=\\"-8.399999999999999\\">mocked</tspan><tspan x=\\"0\\" y=\\"3.6000000000000005\\">text</tspan><tspan x=\\"0\\" y=\\"15.600000000000001\\">value</tspan></text></g></div>"`;

exports[`render Should works with multiple words on same span Should render correctly 1`] = `"<div><g><rect fill=\\"mocked_color_value\\" class=\\"mocked_css_class_value\\" x=\\"-53\\" y=\\"-50\\" rx=\\"5\\" ry=\\"5\\" width=\\"106\\" height=\\"100\\" transform=\\"translate(0,0) scale(1.0942202738310212)\\"></rect><text class=\\"mocked_css_class_value\\" style=\\"text-anchor: middle; font: 10px sans-serif\\" transform=\\"translate(0,0) scale(1.0942202738310212)\\"><tspan x=\\"0\\" y=\\"-8.399999999999999\\">mocked</tspan><tspan x=\\"0\\" y=\\"3.6000000000000005\\">text</tspan><tspan x=\\"0\\" y=\\"15.600000000000001\\">a value</tspan></text></g></div>"`;

exports[`render Should works with undefined text Should render correctly 1`] = `"<div><g><rect fill=\\"mocked_color_value\\" class=\\"mocked_css_class_value\\" x=\\"-53\\" y=\\"-50\\" rx=\\"5\\" ry=\\"5\\" width=\\"6\\" height=\\"0\\" transform=\\"translate(0,0) scale(1)\\"></rect><text class=\\"mocked_css_class_value\\" style=\\"text-anchor: middle; font: 10px sans-serif\\" transform=\\"translate(0,0) scale(1)\\"></text></g></div>"`;
